spec/solver_spec.ruby

```diff
@@ -0,0 +1,50 @@
+require 'solver'
+
+RSpec.describe Solver do
+  describe '#factorial' do
+    it 'calculates the factorial of a number' do
+      solver = Solver.new
+      result = solver.factorial(5)
+      expect(result).to eq(120)
+    end
+    it 'returns 1 for factorial of 0' do
+      solver = Solver.new
+      result = solver.factorial(0)
+      expect(result).to eq(1)
+    end
+  end
```

Comment:
- [ ] The `#factorial` method should have tests that handle negative values which raises an exception as an expectation